wormhole_attester/program/src/attest.rs

```diff
@@ -224,53 +218,70 @@ pub fn attest(ctx: &ExecutionContext, accs: &mut Attest, data: AttestData) -> So
                 ProgramError::InvalidAccountData
             })?;
 
-        // prev_publish_time is picked if the price is not trading
-        let last_trading_publish_time = match price_struct.agg.status {
+        let new_last_attested_trading_publish_time = match price_struct.agg.status {
             PriceStatus::Trading => price_struct.timestamp,
             _ => price_struct.prev_timestamp,
         };
 
-        // Take a mut reference to this price's metadata
-        let state_entry: &mut AttestationState = accs
-            .attestation_state
-            .entries
-            .entry(*price.key)
-            .or_insert(AttestationState {
-                // Use the same value if no state
-                // exists for the symbol, the new value _becomes_ the
-                // last attested trading publish time
-                last_attested_trading_publish_time: last_trading_publish_time,
-            });
-
         let attestation = PriceAttestation::from_pyth_price_struct(
             Identifier::new(price.key.to_bytes()),
             attestation_time,
-            state_entry.last_attested_trading_publish_time, // Used as last_attested_publish_time
+            // Used as last_attested_publish_time, defaults to the new_* value if no pre-existing state is available
+            state
+                .0
+                 .1
+                .last_attested_trading_publish_time
+                .unwrap_or(new_last_attested_trading_publish_time),
             price_struct,
         );
 
-
-        // update last_attested_publish_time with this price's
-        // publish_time. Yes, it may be redundant for the entry() used
-        // above in the rare first attestation edge case.
-        state_entry.last_attested_trading_publish_time = last_trading_publish_time;
-
-        // The following check is crucial against poorly ordered
-        // account inputs, e.g. [Some(prod1), Some(price1),
-        // Some(prod2), None, None, Some(price)], interpreted by
-        // earlier logic as [(prod1, price1), (prod2, price3)].
-        //
-        // Failing to verify the product/price relationship could lead
-        // to mismatched product/price metadata, which would result in
-        // a false attestation.
-        if attestation.product_id.to_bytes() != product.key.to_bytes() {
-            trace!(&format!(
-                "Price's product_id does not match the pased account (points at {:?} instead)",
-                attestation.product_id
-            ));
-            return Err(ProgramError::InvalidAccountData.into());
+        // Update the on-chain value using publish_time or
+        // prev_publish_time if the price is not currently trading
+        state.0.last_attested_trading_publish_time = Some(new_last_attested_trading_publish_time);
+
+        // Serialize the state to calculate rent/account size adjustments
+        let state_serialized = state.0 .1.try_to_vec()?;
+
+        if state.0.is_initialized() {
```

Comment:
I think you probably don't need this resize logic. Can't you make solitaire take care of this?